client/my-sites/plans/current-plan/current-plan-thank-you-card/paid-plan-thank-you-card.js

```diff
@@ -1,97 +1,206 @@
-									<img
-										className="current-plan-thank-you-card__illustration"
-										alt=""
-										aria-hidden="true"
-										src="/calypso/images/illustrations/fireworks.svg"
-									/>
-									<h1 className="current-plan-thank-you-card__title">
-										{ translate( 'Thank you for your purchase!' ) }
-									</h1>
-									<p>{ translate( "Now let's make sure your site is protected." ) }</p>
-									<p>
-										{ translate(
-												"We're setting up spam filters and site backups for you first. Once that's done, our security checklist will guide you through the next steps."
-										) }
-									</p>
-
-									<ProgressBar isPulsing total={ 100 } value={ progressComplete || 0 } />
+				{ installState === INSTALL_STATE_UNCOMPLETE && (
+					<ThankYouCard
+						illustration={ fireworksIllustration }
+						showHideMessage
+						title={ translate( 'Thank you for your purchase!' ) }
+					>
+						<p>{ translate( "Now let's make sure your site is protected." ) }</p>
+						<p>
+							{ preventWidows(
+								translate(
+									"We're setting up spam filters and site backups for you first. Once that's done, our security checklist will guide you through the next steps."
+								)
+							) }
+						</p>
 
-									<p>
-										<a href={ this.getMyPlanRoute() }>{ translate( 'Hide message' ) }</a>
-									</p>
-								</Fragment>
+						<ProgressBar isPulsing total={ 100 } value={ installProgress || 0 } />
+					</ThankYouCard>
+				) }
+				{ installState === INSTALL_STATE_COMPLETE && (
+					<ThankYouCard
+						illustration={ securityIllustration }
+						showContinueButton
+						title={ translate( 'So long spam, hello backups!' ) }
+					>
+						<p>
+							{ preventWidows(
+								translate( 'We’ve finished setting up spam filtering and backups for you.' )
 							) }
-						</div>
-					</Card>
+							<br />
+							{ preventWidows(
+								translate( "You're now ready to finish the rest of the checklist." )
+							) }
+						</p>
+					</ThankYouCard>
 				) }
 			</Fragment>
 		);
 	}
 }
 
 export default connect( state => {
+	const site = getSelectedSite( state );
 	const siteId = getSelectedSiteId( state );
 
+	const installProgress = getJetpackProductInstallProgress( state, siteId );
+
+	let installState;
+	// @TODO we'll need a way to detect generic error states here and add `INSTALL_STATE_ERRORED`
+	if ( installProgress === 100 ) {
+		installState = INSTALL_STATE_COMPLETE;
+	} else {
+		installState = INSTALL_STATE_UNCOMPLETE;
+	}
+
 	return {
-		progressComplete: getJetpackProductInstallProgress( state, siteId ),
-		siteSlug: getSelectedSiteSlug( state ),
+		fileModDisableReason: getSiteFileModDisableReason( site, 'modifyFiles' ),
+		hasMinimumJetpackVersion: site.hasMinimumJetpackVersion,
+		installProgress,
+		installState,
+		isSiteMainNetworkSite: isJetpackSiteMainNetworkSite( state, siteId ),
+		isSiteMultiSite: isJetpackSiteMultiSite( state, siteId ),
+		site,
```

Comment:
Notes on all of this:

- `fileModDisableReason` shouldn't be used because it's not a selector and has the potential to return a new object on every invocation. We could just move it, but I think there's a better solution.
- `hasMinimumJetpackVersion` is a selector that's added to the `site` object by the site selector. I don't really like that magic, but it's the way things work.
- `isSiteMainNetworkSite` and `isSiteMultiSite`- I think we can get rid of these.

Here's the relevant section:

https://github.com/Automattic/wp-calypso/blob/bafb7c5a0b53aa302d2a50708dadb96c086cc822/client/state/sites/selectors.js#L122-L135

Which is merged into `site`:

https://github.com/Automattic/wp-calypso/blob/bafb7c5a0b53aa302d2a50708dadb96c086cc822/client/state/sites/selectors.js#L106-L110

I'd propose:

- remove `fileModDisableReason` and use `site.canUpdateFiles` in `render`
- remove `hasMinimumJetpackVersion` and use `site.hasMinimumJetpackVersion` in `render`
- remove `isSiteMainNetworkSite` and `isSiteMultiSite` and use `site.isSecondaryNetworkSite` in `render`.

Reply:
Good digging, sir! Changed in 32aeef02c2

Reply:
Looks like secondary multisites also get `canUpdateFiles:false` so with previous selectors change, we weren't catching them anymore and were showing "cannot modify files" error to them, too.

So secondary multisites have:
```
isSecondaryNetworkSite: true
canUpdateFiles: false
```

I switched multisite check be before unwriteable files and it works again.